In [1]:
from __future__ import print_function
import os
import sys
import numpy as np


In [2]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0  11193      0 --:--:-- --:--:-- --:--:-- 11228


In [3]:
!mv times.txt $PUIDATA

In [4]:
print("PUIDATA" , os.getenv("PUIDATA"))

PUIDATA /nfshome/vvt221/PUIdata


In [5]:
times = np.loadtxt("%s/times.txt"%os.getenv("PUIDATA"))


## Null Hypothesis Description

 $H_{0}$ : The commute time is on average the same or longer with the new bus route as it was before:
               $T_{new}$ >= $T_{old}$
 $H_{a}$ : The commute time is shorter with the new bus route than it was before: 
              $T_{new}$ < $T_{old}$

In [6]:
population_mean = 36
population_std = 6
alpha_level = 0.05
print("Population Parameters")
print("Population mean: "+ str(population_mean))
print("Population standard deviation: "+ str(population_std))

Population Parameters
Population mean: 36
Population standard deviation: 6


In [7]:
sample_mean = times.mean()
sample_std = times.std()
sample_size = times.size
print("Sample Statistics")
print("Sample mean: "+ str(sample_mean))
print("Sample standard deviation: "+ str(sample_std))
print("Sample Size: "+ str(sample_size))

Sample Statistics
Sample mean: 34.4661616883
Sample standard deviation: 7.10150406819
Sample Size: 100


In [8]:
def calculateZscore(population_mean, sample_mean, population_std, sample_size ):
    return ((population_mean - sample_mean)/(population_std/(sample_size ** 0.5)))

In [9]:
zscore = calculateZscore(population_mean, sample_mean, population_std, sample_size )
zscore

2.5563971861666701

### Zscore of our confidence interval = 1.65

### Following observerations have been made:
###  Zscore of our sample > Zscore of our population interval
### Since, Zscore = 2.556 , ie , more than 2 standard deviations 
### Thus, we reject the null hypothesis at the 0.05 significance level